src/irmin-pack/pack_value_intf.ml

```diff
@@ -1,5 +1,13 @@
 open! Import
 
+(** This module defines abstractions over entries in the pack file, which are
+    encoded as the following sequence:
+
+    - the (fixed-length) hash of the data stored in this entry;
+    - the {i kind} of data being stored (contents, nodes, blob etc.);
+    - the data itself, optionally with a length header that contains the encoded
+      size of the data section (excluding the header itself). *)
+
```

Comment:
Very useful comment. Thanks!

Reply:
Is there any way to detect if the "data itself" has a length header or not? Is this indicated by the "kind" (contents_with_length, contents_without_length etc)? Or do we just not know, in general? Do we plan to use length headers always in future (where the kind of data doesn't relate to a fixed-size thing)?

Reply:
`Pack_value.Kind.length_header_exn` knows this

Reply:
How can it distinguish properly real length headers from contents data that starts off looking like a length header?

Reply:
But for some kinds, the length_header will not know whether there is or is not?

Reply:
At compile time, we know the header structure of all possible `kind`s except the contents value, since this is supplied by the user. At functor instantiation time, the picture is completed by `Conf.contents_length_header`.

> Do we plan to use length headers always in future (where the kind of data doesn't relate to a fixed-size thing)?

At least with the current pack entry format (with a compulsory hash field), it seems worth always having length headers for efficiency reasons. We still support the V0 objects w/o length headers out there in the wild though.